packages/odf/modals/StorageClientDistributionModal/StorageClientAttacherComponent.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import { StorageConsumerModel } from '@odf/core/models';
+import {
+  getName,
+  ModalFooter,
+  RowComponentType,
+  SelectableTable,
+  StorageClassModel,
+  StorageClassResourceKind,
+  StorageConsumerKind,
+  useCustomTranslation,
+  VolumeSnapshotClassKind,
+  VolumeSnapshotClassModel,
+} from '@odf/shared';
+import useInterceptSubmit from '@odf/shared/hooks/useMouseEventListener';
+import { CommonModalProps } from '@odf/shared/modals';
+import { sortRows } from '@odf/shared/utils';
+import {
+  K8sModel,
+  k8sPatch,
+  ListPageFilter,
+  useK8sWatchResource,
+  useListPageFilter,
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { ModalComponent } from '@openshift-console/dynamic-plugin-sdk/lib/app/modal-support/ModalProvider';
+import { Button, Flex, FlexItem, Modal } from '@patternfly/react-core';
+import { Td } from '@patternfly/react-table';
+import { generatePatchForDistributionOfResources } from '../../components/ResourceDistribution/utils';
+import { LastHeartBeat } from '../../components/storage-consumers/client-list';
+import './storageClientAttacherComponent.scss';
+
+const RowComponent: React.FC<RowComponentType<StorageConsumerKind>> = ({
+  row: client,
+}) => {
+  const name = getName(client);
+  const clientClusterId = client?.status?.client?.clusterId;
+  const clusterName = `${client?.status?.client?.clusterName || '-'} (${
+    clientClusterId || '-'
+  })`;
```

Comment:
nits (I guess `client` itself can never be undefined or null):
```suggestion
  const clientClusterId = client.status?.client?.clusterId;
  const clusterName = `${client.status?.client?.clusterName || '-'} (${
    clientClusterId || '-'
  })`;
```